pkg/clusters/utils.go

```diff
@@ -75,3 +75,30 @@ func GetIngressLoadbalancerStatus(ctx context.Context, c Cluster, namespace stri
 		return nil, fmt.Errorf("%T is not a supported ingress type", ingress)
 	}
 }
+
+// CreateNamespace create customized namespace
+func CreateNamespace(ctx context.Context, cluster Cluster, namespace string) error {
+	nsName := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	}
+
+	_, err := cluster.Client().CoreV1().Namespaces().Create(context.Background(), nsName, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed creating namespace %s, err %v", namespace, err)
+	}
+
+	nsList, err := cluster.Client().CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed creating namespace: %w", err)
+	}
+	for _, item := range nsList.Items {
+		if item.Name == namespace && item.Status.Phase == corev1.NamespaceActive {
+			fmt.Printf("created namespace %s successfully.", namespace)
+			return nil
+		}
+	}
```

Comment:
There is technically a race condition here, we should add some retry logic OR watch logic to ensure we don't retrieve the namespace during a time window where the status phase is not yet updated, otherwise we would fail here when we should have succeeded.

Reply:
Added retry 
https://github.com/Kong/kubernetes-testing-framework/pull/104/files#diff-e7bf152c56db96b7d769fe6e6ad4cc4ad16e1c1a42b5558eb793fa38979b1877R89